### ISSUE TITLE
The quantum vs classical comparison is not parameter-matched: no classical-equivalent (SE-block) ablation

### ISSUE THREAD

**Issue by Kuo-TingKai** [ChiShengChen/Quantum_MNIST_Diffusion#3]
## Summary

The classical vs quantum comparison in the README is not parameter-matched or architecture-matched. Setting `use_quantum=True` adds a module that has **no counterpart at all** in the `use_quantum=False` baseline, so the comparison cannot isolate the contribution of the quantum circuit.

## What the flag actually changes

```python
if use_quantum:
    self.q_attn = QuantumLayer()
...
if self.use_quantum:
    q_weight = self.q_attn(pooled).unsqueeze(-1).unsqueeze(-1)
    x = x * q_weight
```

`use_quantum=False` has **no bottleneck gating of any kind**. So the ablation being run is

> U-Net  vs.  U-Net + (128→16 projection → nonlinearity → 16→128 projection → channel-wise multiplicative gate)

which is structurally a **Squeeze-and-Excitation block** (Hu et al., CVPR 2018) with reduction ratio 8, where the QNode sits in the place of SE's `ReLU`/`sigmoid`. SE blocks are well documented to improve conv nets on their own. Any difference measured this way is confounded between "quantum" and "channel attention exists at all".

Given #1 and #2, the confound is not hypothetical: in v7 the quantum branch contributes a fixed feature map plus one trainable `nn.Linear(16, 128)` that the baseline lacks.

## Proposed ablation ladder

I think this is the single change that would most strengthen the paper, because "you didn't compare against the obvious classical equivalent" is the standard reviewer objection to quantum-advantage claims. Concretely, four arms sharing every other hyperparameter, seed and data split:

| arm | bottleneck module | trainable params in module |
|---|---|---|
| **A** plain | none (current `use_quantum=False`) | 0 |
| **B** classical SE | `Linear(128,16) → ReLU → Linear(16,128)` | 2 304 |
| **C** frozen-random SE | same as B, but the 128→16 projection frozen at init | 2 048 |
| **D** quantum | current `QuantumLayer` | 2 304 + 48 |

* **B vs D** is the honest test of the quantum circuit.
* **C** matters specifically because of `.detach()`: it reproduces the "encoder is a fixed random projection" condition, so if D ≈ C the gain is attributable to the random projection, not the circuit.
* **A vs B** tells you how much of the reported effect is just "SE gating helps".

Reporting the parameter count (and ideally FLOPs / wall-clock) per arm in the results table would also help.

## Related

A second, cheaper matching issue: the two arms are also not matched on *effective capacity in the decoder*. In v7, `use_quantum` changes only the bottleneck, which is fine — but the README's v8 table reports quantum *worse* than classical on the full skip-connection U-Net, which is exactly what you'd expect if the "advantage" were an artifact of the weaker v7 bottleneck rather than of the circuit. Running the ladder above on both v7 and v8 would settle that.

**Comment by ChiShengChen** [comment]
Implemented in #9. You were right that `plain` is not a control — `use_quantum=False` had no bottleneck module at all.

The four arms are in `bottlenecks.py`, selected with `--arm`. Measured counts at `h=16`:

| arm | bottleneck trainable | total |
|---|---|---|
| `plain` | 0 | 0 |
| `se` | 4 240 | 4 240 |
| `se_frozen` | 2 176 | 4 240 |
| `quantum` | 4 336 | 4 336 |

`se` and `quantum` differ by exactly **96** parameters — the circuit's own weights (96, not 48, after #7 doubled them). So `se` vs `quantum` isolates the circuit. There's a test asserting that invariant, because if the two arms silently drift apart the confound this ladder exists to remove comes straight back.

Two deviations from your spec worth flagging:

1. **The gate is a plain multiplication, not a sigmoid.** The quantum arm already did `x = x * q_weight` unbounded, so adding a sigmoid to the classical arm only would have made them differ in how the gate is bounded as well as in the bottleneck. Matching the existing (unbounded) behaviour keeps the single-variable comparison.
2. **Your parameter estimate for B was 2 304; the actual is 4 240.** `Linear(128,16)` + `Linear(16,128)` is 2 064 + 2 176 with biases. Doesn't change the argument, just the table.

Scope: v7 MNIST only. pathmnist and v8 need the same treatment before the ladder can run on them, which is the part of this issue I haven't done.

**Comment by ChiShengChen** [comment]
Ported to v8 and pathmnist, so the ladder now covers all three models (the part of this issue I'd left open).

**v8 needed a different classical control than I first assumed.** Its `QuantumLayer` has no `input_proj` — the circuit reduces its own input with `adaptive_avg_pool1d`, so the quantum arm's only classical parameters are `output_proj`. Dropping v7's `SEBottleneck` in as the control would have given the classical arm 2 064 parameters the quantum arm doesn't have, i.e. re-created this exact confound in the thing meant to remove it. `SEBottleneck` now takes `down="linear"|"pool"` and v8 uses pooling.

Verified bottleneck parameter counts at `n_hidden=16`:

| script | `plain` | `se` | `se_frozen` | `quantum` | quantum − se |
|---|---|---|---|---|---|
| v7 mnist | 0 | 4 240 | 2 176 | 4 336 | **96** |
| pathmnist | 0 | 4 240 | 2 176 | 4 336 | **96** |
| v8 (pool down) | 0 | 2 176 | rejected | 2 272 | **96** |

96 is the circuit's weight count in every case, so `se` vs `quantum` isolates the circuit in all three. There's a per-script test asserting it.

`se_frozen` **does not apply to v8** — pooling has no parameters to freeze, so the arm would be a silent duplicate of `se`. It raises there instead, the v8 CLI omits it from `--arm`'s choices, and `run_scaling_study.py` drops it with a note. Your ladder is 4 arms on v7/pathmnist and 3 on v8.

One caveat on the "run the ladder on both v7 and v8" plan: **only v7 has a step budget.** v8 and pathmnist still train a fixed number of epochs, so their cells at different `--n-train` aren't compute-matched — a curve built that way conflates "less data" with "fewer updates". The runner warns rather than letting you produce it. Porting the step budget to v8 is the remaining piece if you want the scaling curve on it too.

Also fixed two things that would have bitten the sweep: `load_pathmnist` was building its label index by pulling all ~90k images through Resize + ToTensor + Normalize just to read labels, and both v8 and pathmnist built their EMA copy with `copy.deepcopy`, which raises on any non-default simulator — so `--qdevice lightning.qubit` would have been unusable on them.
